PSIni/Private/Save-WithCleanup.ps1

```diff
@@ -10,14 +10,18 @@ function Remove-EmptyLines {
         [Parameter( Mandatory )]
         [ValidateNotNullOrEmpty()]
         [String]
-        $FilePath,
+        $Path,
 
-        $Force
+        $Force,
+
+        [Parameter( Mandatory )]
+        [Object[]]
+        $FileContent
     )
 
     $lf = if ((-not $PSVersionTable.Platform) -or ($PSVersionTable.Platform -eq "Win32NT")) { "\r\n" }
     else { "\n" }
 
-    $content = (Get-Content -Path $FilePath -Raw) -replace "(${lf})*$", "" -replace "(${lf}){3,}", ""
-    Set-Content -Value $content -Path $FilePath -Encoding $Encoding -Force
+    $content = $FileContent -replace "(${lf})*$", "" -replace "(${lf}){3,}", ""
+    Set-Content -Value $content -Path $Path -Encoding $Encoding -Force
```

Comment:
@lipkau 
I am considering replacing `Set-Content` with `Out-File`, because:
- It doesn't block reading the target file
- It allows appending content to the file, without reading its contents to memory

on the other hand, if we are going to be swapping around Encoding & lineEnds, maybe it is better to read the entire targetFile into memory & replacing it in its entirety? What do you think @lipkau 

Reference about `Set-Content` & `Out-File`: https://stackoverflow.com/a/13544067

Reply:
reading to memory, processing and saving with `Out-File` seems appropriate to me.
I have opted for `Out-File` over `Set-Content` before.

I need to remember why I added the changing of line endings, tho 😕